File: app/src/main/java/ru/laushkina/rates/repository/RateEntity.java
package ru.laushkina.rates.repository;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Random;

import ru.laushkina.rates.model.Rate;

@Entity( tableName = "rates")
public class RateEntity {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String shortName;
    public Float amount;
    public boolean isBase;

    public RateEntity(int id, @NonNull String shortName, @NonNull Float amount, boolean isBase) {
        this.id = id;
        this.shortName = shortName;
        this.amount = amount;
        this.isBase = isBase;
    }

    public RateEntity(@NonNull Rate rate) {
        this(new Random().nextInt(), rate.getShortName(), rate.getAmount(), rate.isBase());
    }
}

File: app/src/main/java/ru/laushkina/rates/repository/RateEntity.kt
package ru.laushkina.rates.repository

import androidx.room.Entity
import androidx.room.PrimaryKey
import ru.laushkina.rates.model.Rate
import java.util.*

@Entity(tableName = "rates")
class RateEntity(
        @PrimaryKey(autoGenerate = true) val id: Int,
        public val shortName: String,
        val amount: Float,
        val isBase: Boolean) {

    constructor(rate: Rate) : this(Random().nextInt(), rate.shortName.name, rate.amount, rate.isBase)
}